activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -27,6 +27,8 @@ def sql(event)
 
       return if IGNORE_PAYLOAD_NAMES.include?(payload[:name])
 
+      pool_name = payload[:connection]&.pool&.pool_config&.db_config&.name
+      pool_info = pool_name.present? && pool_name != "primary" ? "[#{pool_name}] " : ""
```

Comment:
The pool name in Rails 6.1 for the default pool is `ActiveRecord::Base` and is no longer `primary`. We changed this in https://github.com/rails/rails/pull/38190. I don't think we should do any check for `pool_name != "primary"`, just always add the pool_name.